app/models/controlled_vocabulary.rb

```diff
@@ -1,75 +1,123 @@
 # frozen_string_literal: true
+
+# Base Class for controlled vocabularies for local authorities
+# Terms within these vocabularies are stored within YAML files
+# *Please* note that these are distinct from EphemeraVocabulary and Vocab Classes
 class ControlledVocabulary
   class_attribute :handlers
   self.handlers = {}
+
+  # Mutates the Class attribute mapping a key symbol to a ControlledVocabulary subclass
+  # @param key [Symbol] the symbolized name of the registered vocabulary
+  # @param klass [Class] the ControlledVocabulary subclass
   def self.register(key, klass)
     handlers[key] = klass
   end
 
+  # Accesses the Class mapped to a key symbol for a ControlledVocabulary subclass
+  # @param key [Symbol] the symbolized name for the registered vocabulary
+  # @return [Class] the ControlledVocabulary subclass
   def self.for(key)
     (handlers[key] || self).new(key: key)
   end
 
+  # Initializer for the ControlledVocabulary
   attr_reader :key
+  # @param key [Symbol] the symbolized name for the vocabulary
   def initialize(key: nil)
     @key = key
   end
 
+  # Method returning all terms within the vocabulary
+  # This should be overridden
   def all
     []
   end
 
+  # Accesses a Term Object matching a specific value
+  # @param value [String] the value for the Term being searched for
+  # @return [Term] the Term Object within the vocabulary
   def find(value)
     all.find { |x| x.value == value }
   end
 
+  # Method which determines whether or not a Term Object containing a value exists in this vocabulary
+  # @param value [String] the value for Term
+  # @return [Boolean] whether or not the Term is included in this vocabulary
   def include?(value)
     !find(value).nil?
   end
 
+  # Base Class for terms within a ControlledVocabulary
+  # Please note that this Class does *not* contain a property for term ownership
   class Term < Valkyrie::Resource
     attribute :label, Valkyrie::Types::String
     attribute :value, Valkyrie::Types::Any
     attribute :notable, Valkyrie::Types::Bool
     attribute :definition, Valkyrie::Types::String
+
+    # Whether or not this Term has been marked as notable
+    # @return [Boolean] whether or not this Term is notable
```

Comment:
what does it mean to be notable?

Reply:
Whether or not a general note/description field should be shown with this term.  This is used in Plum to trigger the display of a rights note for some rights statements but not others.

Reply:
okay, but the note itself is fixed / not editable? And I guess these are pulled from locales.

Reply:
In Plum, there is a rights note field that is hidden or displayed on the edit form based on which rights statement is selected.  So if you select a rights statement that is "notable", the rights note input box is shown so the user can enter a rights note, and if you select a rights statement that isn't "notable", the note field is cleared and hidden.